aten/src/ATen/native/sparse/SparseTensor.cpp

```diff
@@ -230,22 +227,28 @@ Tensor sparse_coo_tensor(const Tensor& indices, const Tensor& values_, ArrayRef<
                "size is inconsistent with indices: for dim ", d, ", size is ", dim_size, " but found index ", max_index_in_dim);
     }
   }
+  return true;
+}
 
-  return at::_sparse_coo_tensor_with_dims_and_tensors(
-      sparse_dim, dense_dim, size, indices, values, values.options().layout(kSparse));
+// NB: Got rid of the sizes == NULL case
+Tensor sparse_coo_tensor(const Tensor& indices, const Tensor& values_, ArrayRef<int64_t> size, const TensorOptions& options) {
+  // arg checking
+  TORCH_CHECK(!options.has_layout() || options.layout() == kSparse, "expected sparse layout, but got layout ", options.layout());
+
+  Tensor values = expand_values_if_needed(values_);
```

Comment:
I don't think this is necessary here OR you should remove the `expand_values_if_needed` from the validate/unsafe calls below.